### ISSUE TITLE
Hello. Can you share your implementations for Imagenet?

### ISSUE THREAD

**Issue by wjun0830** [hankook/SLA#2]
Hello.

Can you share your implementations for ImageNet? 
I saw you used the CutMix for baseline but it would be great help if you provide the merged version!

**Comment by hankook** [comment]
Thank you for your interest! The ImageNet data loading code is already implemented [here](https://github.com/hankook/SLA/blob/master/datasets.py). You can train ResNets with ImageNet using `--dataset` and `--datadir` options. As I mentioned in [README](https://github.com/hankook/SLA/blob/master/README.md), using the `--with-large-loss` option might be helpful for training.

For CutMix, we simply modify [the official implementation](https://github.com/clovaai/CutMix-PyTorch). I recommend seeing [SLA/trainers.py:L127-L145](https://github.com/hankook/SLA/blob/master/trainers.py#L127-L145) and [CutMix/train.py:L224-L244](https://github.com/clovaai/CutMix-PyTorch/blob/master/train.py#L224-L244) codes for the modification.

**Comment by wjun0830** [comment]
Thanks for help!

**Comment by wjun0830** [comment]
Oh one more question btw...

You indicated that you have trained with batch_size 128.
Is it right that actual batch size is 512 when using rotation for augmentation?
There was no indication in the paper but it seems like that in your implementation

**Comment by hankook** [comment]
Yes, you're right. Note that we mentioned about the implementation detail in our paper (see page 3) as follows:
> During training, we feed all M augmented samples simultaneously for each iteration as Gidaris et al. (2018) did, i.e., ...

**Comment by wjun0830** [comment]
Oh right Thanks.
And I am currently testing your model on tiny-imagenet.
Since your link in the paper is not accessible, I downloaded tiny-imagenet on my own.

However, when I trained your model on ResNet18 with Tiny-imagenet, its accuracy on val and test is very poor. It outputs around 0.0058% which is very random.

The only part I changed in your implementation is the data root which was 'val2' and I changed it to 'val'.

Below is an example of my running-log.
```
2021-04-19 07:10:29,764] [test] [iteration 4000] [total_loss 21.0402] [total_acc 0.0015] [single_loss 10.2201] [single_acc 0.0022] [agg_loss 9.6209] [agg_acc 0.0030]
```

This issue occurs until the end of training.

Training loss and acc seems fine and I think your validation set has different structure. Maybe can you share that?

**Comment by hankook** [comment]
ImageFolder is working only if sub-directories in the data root directory should be categories as described [here](https://pytorch.org/vision/stable/datasets.html#imagefolder). Could you share your `val` folder structure? If what I remember is right, the initial `val` folder does not contain any sub-directory and all `val` images are directly contained in the folder.

**Comment by wjun0830** [comment]
Sorry for bothering much but cannot generate your baseline for tiny-imagenet dataset.

Oops...I used different resolution tiny-imagenet haha sorry for my mistake.
Did you use (bsz 128), (lr 0.1) as you trained your SLA?

**Comment by hankook** [comment]
Yes we did. We also used standard ResNet architecture for non-CIFAR datasets, i.e., we used `--model resnet18`. Moreover, as you mentioned, our tiny-ImageNet images had the same resolution of original ImageNet ones. Since some tiny-ImageNet dataset has 64x64 resolution, the results could be different from the results reported in our paper when using such low-resolution images. 

Do you solve the dataset issue? I'm currently finding the tiny-ImageNet ... but I'm not sure that there still is the data in our servers.

**Comment by wjun0830** [comment]
Ours seems to be 64x64.
Its size of single image is only around 2.0 KB while the image size of Imagenet is usually over 80KB.

I downloaded the dataset from the command 'wget stanford 231n.....'.

Is there other site that publish tiny-imagnet in 224x224?

**Comment by hankook** [comment]
I'm not sure ... When I experimented with tiny-ImageNet, I downloaded the data from "https://tiny-imagenet.herokuapp.com". If your goal is to obtain reasonable results in tiny-ImageNet (not perfectly reproduce our results), then I recommend to modify crop size from 224/256 to 96 (see [datasets.py:L29-L38](https://github.com/hankook/SLA/blob/master/datasets.py#L29-L38)) and use `--model cresnetXX` (`XX` is the depth, e.g., `32`, `56`, ...) because standard ResNet architecture (`resnetXX`) reduces the resolution from 64x64 to 16x16 at the first two layers (conv1 and maxpool1), which could degrade classification performance with a large margin.

**Comment by wjun0830** [comment]
Oh alright. I better try other way. Tiny 224x224 is so hard to obtain haha
Maybe I will try with 64x64 tiny without maxpool layer in ResNet18.
Thanks for help and plz let me know if you still have 224x224 dataset!!
